Сначала откроем выбранный датасет. Загрузим таблицу в переменную data и выведем ее. Укажем, что разделителем по столбцам в csv файле является запятая

In [ ]:
import pandas as pd
data = pd.read_csv('bmi.csv', delimiter = ',')

In [6]:
import pandas as pd
data = pd.read_csv('bmi.csv', delimiter = ',')
print(data)

     Gender  Height  Weight  Index
0      Male     174      96      4
1      Male     189      87      2
2    Female     185     110      4
3    Female     195     104      3
4      Male     149      61      3
..      ...     ...     ...    ...
495  Female     150     153      5
496  Female     184     121      4
497  Female     141     136      5
498    Male     150      95      5
499    Male     173     131      5

[500 rows x 4 columns]


Выведем первый и третий столбцы выборки сначала вместе, потом по отдельности.
Функция iloc позволяет выбирать объекты таблицы по индексам. Первый аргумент - индекс или срез строк, второй - индекс или срез столбцов

In [7]:
print(data.iloc[:, 0:4:2])
print(data.iloc[:,0])
print(data.iloc[:,2])

     Gender  Weight
0      Male      96
1      Male      87
2    Female     110
3    Female     104
4      Male      61
..      ...     ...
495  Female     153
496  Female     121
497  Female     136
498    Male      95
499    Male     131

[500 rows x 2 columns]
0        Male
1        Male
2      Female
3      Female
4        Male
        ...  
495    Female
496    Female
497    Female
498      Male
499      Male
Name: Gender, Length: 500, dtype: object
0       96
1       87
2      110
3      104
4       61
      ... 
495    153
496    121
497    136
498     95
499    131
Name: Weight, Length: 500, dtype: int64


Выведем первые 10 строк выборки, используя тот же метод iloc

In [10]:
print(data.iloc[:10, :])

   Gender  Height  Weight  Index
0    Male     174      96      4
1    Male     189      87      2
2  Female     185     110      4
3  Female     195     104      3
4    Male     149      61      3
5    Male     189     104      3
6    Male     147      92      5
7    Male     154     111      5
8    Male     174      90      3
9  Female     169     103      4


Выведем 23 случайных строки из выборки, используя метод sample

In [9]:
print(data.sample(n = 23))

     Gender  Height  Weight  Index
108  Female     181     106      4
431  Female     163     154      5
466  Female     188      99      3
423  Female     178     107      4
248  Female     188      90      3
464  Female     181      80      2
125    Male     158      96      4
384  Female     182     149      5
140  Female     169      54      2
424    Male     174      54      1
193  Female     142     137      5
355    Male     174      90      3
373  Female     163      63      2
386    Male     146     138      5
397    Male     169     136      5
277    Male     190      80      2
460    Male     179     123      4
376    Male     142      91      5
50   Female     168     135      5
471    Male     154     112      5
70   Female     153      51      2
287    Male     178     160      5
100    Male     199     156      4


Выведем статистические данные из числовых столбцов (последние 3). Используя метод describe()

In [11]:

h = data['Height']
w = data['Weight']
i = data['Index']
print(f'Столбец "рост": {h.describe()}')
print(f'Столбец "вес: {w.describe()}')
print(f'Столбец "индекс": {i.describe()}')

Столбец "рост": count    500.000000
mean     169.944000
std       16.375261
min      140.000000
25%      156.000000
50%      170.500000
75%      184.000000
max      199.000000
Name: Height, dtype: float64
Столбец "вес: count    500.000000
mean     106.000000
std       32.382607
min       50.000000
25%       80.000000
50%      106.000000
75%      136.000000
max      160.000000
Name: Weight, dtype: float64
Столбец "индекс": count    500.000000
mean       3.748000
std        1.355053
min        0.000000
25%        3.000000
50%        4.000000
75%        5.000000
max        5.000000
Name: Index, dtype: float64


выведем разницу в статистическом поле в данных для женщин и мужчин, так как на основе этих данных можно будет сделать некоторые выводы

In [12]:

men = data[data['Gender'] == 'Male']
women = data[data['Gender'] == 'Female']
print(f'Статистика для мужчин: {men[['Height', 'Weight', 'Index']].describe()}')
print(f'Статистика для женщин: {women[['Height', 'Weight', 'Index']].describe()}')

Статистика для мужчин:            Height      Weight       Index
count  245.000000  245.000000  245.000000
mean   169.648980  106.314286    3.787755
std     17.065665   31.832734    1.386355
min    140.000000   50.000000    0.000000
25%    154.000000   80.000000    3.000000
50%    171.000000  105.000000    4.000000
75%    183.000000  137.000000    5.000000
max    199.000000  160.000000    5.000000
Статистика для женщин:            Height      Weight       Index
count  255.000000  255.000000  255.000000
mean   170.227451  105.698039    3.709804
std     15.711825   32.962025    1.325886
min    140.000000   50.000000    0.000000
25%    157.000000   79.000000    3.000000
50%    170.000000  106.000000    4.000000
75%    184.000000  135.000000    5.000000
max    199.000000  160.000000    5.000000


Сделаем базовую группировку по полу

In [15]:
grouped_by_gender = data.groupby('Gender').agg({
    'Height': ['mean', 'std', 'min', 'max'],
    'Weight' : ['mean', 'std', 'min', 'max'],
    'Index': ['mean', 'std', 'min', 'max']
})
print(f'Статистика по полу: {grouped_by_gender}')

Статистика по полу:             Height                           Weight                      \
              mean        std  min  max        mean        std min  max   
Gender                                                                    
Female  170.227451  15.711825  140  199  105.698039  32.962025  50  160   
Male    169.648980  17.065665  140  199  106.314286  31.832734  50  160   

           Index                    
            mean       std min max  
Gender                              
Female  3.709804  1.325886   0   5  
Male    3.787755  1.386355   0   5  


**Выводы: **
1. Антопометрические данные мужчин и женщин в данной выборке почти идентичны:
Пример: Средний рост женщин = 170
Средний рост мужчин = 169
Похожая ситуация с весом и имт
Это говорит о некоторой специфичности данной выборки
2. Средний ИМТ у обоих полов = 3.7, при максимальном значении = 5
Можно сказать, что большая часть выборки состоит из объектов с высоким показателем имт
